pyalgotrading/algobulls/connection.py

```diff
@@ -310,14 +409,21 @@ def backtest(self, strategy=None, start=None, end=None, instruments=None, lots=1
 
         # Sanity checks - Convert config parameters
         _error_msg_candle = f'Argument "candle" should be a valid string or an enum of type CandleInterval. Possible string values can be: {get_valid_enum_names(CandleInterval)}'
-        _error_msg_timestamps = f'Argument "start" should be a valid timestamp string (YYYY-MM-DD | HH:MM) or an instance of type datetime.datetime'
+        _error_msg_timestamps = f'\nString Format (YYYY-MM-DD HH:MM z) or an instance of type datetime.datetime for Back Testing \nString Format (HH:MM z) or an instance of type datetime.time for Real trading or Paper Trading'
         _error_msg_instruments = f'Argument "instruments" should be a valid instrument string or a list of valid instruments strings. You can use the \'get_instrument()\' method of AlgoBullsConnection class to search for instruments'
         _error_msg_mode = f'Argument "mode" should be a valid string or an enum of type StrategyMode. Possible string values can be: {get_valid_enum_names(StrategyMode)}'
 
+        initial_funds_virtual = float(initial_funds_virtual)
         if isinstance(start, str):
-            start = dt.strptime(start, '%Y-%m-%d | %H:%M')
+            try:
+                start = get_datetime_with_tz(start, trading_type.name)
+            except ValueError as ve:
+                raise ValueError(f'Error : Invalid string timestamp format for argument "start" ({ve}).\nExpected timestamp format for {trading_type.name} is [{TRADE_TYPE_DT_FORMAT_MAP[trading_type.name][KEY_DT_ZONE]}]')
         if isinstance(end, str):
-            end = dt.strptime(end, '%Y-%m-%d | %H:%M')
+            try:
+                end = get_datetime_with_tz(end, trading_type.name)
+            except ValueError as ve:
```

Comment:
use `ex` as exception obj name and not `ve`

Reply:
done